R/cheers_checker.R

```diff
@@ -29,20 +29,27 @@ find_next_vector_element <- function(value, vector) {
 #' @description Find the previous element of the vector before a value
 #' @param value A value of numeric values
 #' @param vector A vector of numeric values
+#' @param LTE a boolean to determine collection on "less than" or "less than equal"
 #' @return The previous element of the vector before the value
 #' @export
 #' @examples
 #' \dontrun{
 #' find_previous_vector_element(value = 5, vector = c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10))
 #' }
 #'
-find_previous_vector_element <- function(value, vector){
+find_previous_vector_element <- function(value, vector, LTE=FALSE){
+   
    # Find the elements in the vector that are less than the specified value
-   less_than_value <- vector[vector < value]
+   if(LTE==TRUE) {
+      less_than_value <- vector[vector <= value]
+   } else {
+      less_than_value <- vector[vector < value]
+   }
 
-  # If there are no elements less than the specified value, return the value
+  # If there are no elements less than the specified value, 
+  # return the minimum value found in the vector
    if (length(less_than_value) == 0) {
-     return(value)
+     return (min(vector))
```

Comment:
This element could be greater than the value, so should probably return NA here.